lib/stripe/list_object.rb

```diff
@@ -12,10 +30,35 @@ def [](k)
       end
     end
 
+    # Iterates through each resource in the page represented by the current
+    # `ListObject`.
+    #
+    # Note that this method makes no effort to fetch a new page when it gets to
+    # the end of the current page's resources. See also +auto_paging_each+.
     def each(&blk)
       self.data.each(&blk)
     end
 
+    # Iterates through each resource in all pages, making additional fetches to
+    # the API as necessary.
+    #
+    # Note that this method will make as many API calls as necessary to fetch
+    # all resources. For more granular control, please see +each+ and
+    # +next_page+.
+    def auto_paging_each(&blk)
```

Comment:
@russelldavis I personally really like having the `(&blk)` on the end of the method because it makes it emminently obvious when you're reading through code that this is a method that takes a block. I've changed the implementation to respect that variable in 53b214f as you suggested. Looks very, very clean IMO. Excellent call!